cmd/notation/plugin/install.go

```diff
@@ -0,0 +1,315 @@
+		if opts.inputCheckSum == "" {
+			return errors.New("install from URL requires non-empty SHA256 checksum of the plugin source")
+		}
+		tmpDir, err := os.MkdirTemp("", notationPluginTmpDir)
+		if err != nil {
+			return fmt.Errorf("failed to create notationPluginTmpDir, %w", err)
+		}
+		defer os.RemoveAll(tmpDir)
+		downloadPath, err := notationplugin.DownloadPluginFromURL(ctx, opts.pluginSource, tmpDir)
+		if err != nil {
+			return fmt.Errorf("failed to download plugin from URL %s with error: %w", opts.pluginSource, err)
+		}
+		return installFromFileSystem(ctx, downloadPath, opts.inputCheckSum, opts.force)
+	default:
+		return fmt.Errorf("failed to install the plugin, plugin source type %s is not supported", pluginSourceType)
+	}
+}
+
+// getPluginSource returns the type of plugin source
+func getPluginSource(source string) (notationplugin.PluginSourceType, error) {
+	source = strings.TrimSpace(source)
+	// check file path
+	_, fileError := os.Stat(source)
+	if fileError == nil {
+		return notationplugin.TypeFile, nil
+	}
+	// check url
+	if strings.HasPrefix(source, httpsPrefix) {
+		return notationplugin.TypeURL, nil
+	}
+	// unknown
+	fmt.Fprintf(os.Stdout, "%q is not a valid file: %v\n", source, fileError)
+	fmt.Fprintf(os.Stdout, "%q is not a valid HTTPS URL\n", source)
+	return notationplugin.TypeUnknown, fmt.Errorf("%q is an unknown plugin source. Require file path or HTTPS URL", source)
+}
+
+// installFromFileSystem install the plugin from file system
+func installFromFileSystem(ctx context.Context, inputPath string, inputCheckSum string, force bool) error {
+	// sanity check
+	inputFileStat, err := os.Stat(inputPath)
+	if err != nil {
+		return fmt.Errorf("failed to install the plugin, %w", err)
+	}
+	if !inputFileStat.Mode().IsRegular() {
+		return fmt.Errorf("failed to install the plugin, %s is not a regular file", inputPath)
+	}
+	if inputCheckSum != "" {
+		if err := notationplugin.ValidateCheckSum(inputPath, inputCheckSum); err != nil {
+			return fmt.Errorf("failed to install the plugin, %w", err)
+		}
+	}
+	// install the plugin based on file type
+	fileType, err := osutil.DetectFileType(inputPath)
+	if err != nil {
+		return fmt.Errorf("failed to install the plugin, %w", err)
+	}
+	switch fileType {
+	case notationplugin.TypeZip:
+		if err := installPluginFromZip(ctx, inputPath, force); err != nil {
+			return fmt.Errorf("failed to install the plugin, %w", err)
+		}
+		return nil
+	case notationplugin.TypeGzip:
+		// when file is gzip, require to be tar
+		if err := installPluginFromTarGz(ctx, inputPath, force); err != nil {
+			return fmt.Errorf("failed to install the plugin, %w", err)
+		}
+		return nil
+	default:
+		return errors.New("failed to install the plugin, invalid file format. Only support .tar.gz and .zip")
+	}
+}
+
+// installPluginFromZip extracts a plugin zip file, validates and
+// installs the plugin
+func installPluginFromZip(ctx context.Context, zipPath string, force bool) error {
```

Comment:
In the current code, `zip` and `tar.gz` are sharing the same core process of plugin installation. Since [tar.Reader](https://pkg.go.dev/archive/tar#NewReader) does not implement io/fs.FS, there's going to be many duplicated codes. 
The current core process takes in an `io.Reader`, which is very general and covers both of our cases. So I think we should keep the current way?

Reply:
Using `fs.FS` will make the logic more readable and it opens possibilities in the future like installing plugins from a folder, or `embed.FS`, `fstest.MapFS` for unit testing, etc.

Reply:
Update the code.